qiskit_experiments/library/tomography/fitters/postprocess_fit.py

```diff
@@ -0,0 +1,166 @@
+    states_metadata = []
+    fit_traces = []
+    total_trace = 0.0
+    for i, fit in enumerate(fits):
+        # Get eigensystem of state fit
+        raw_eigvals, eigvecs = _state_eigensystem(fit)
+
+        # Optionally rescale eigenvalues to be non-negative
+        if make_positive and np.any(raw_eigvals < 0):
+            eigvals = _make_positive(raw_eigvals)
+            fit = eigvecs @ (eigvals * eigvecs).T.conj()
+            rescaled_psd = True
+        else:
+            eigvals = raw_eigvals
+            rescaled_psd = False
+
+        # Optionally rescale fit trace
+        fit_trace = np.sum(eigvals).real
+        fit_traces.append(fit_trace)
+        if (
+            trace is not None
+            and not np.isclose(fit_trace, 0, atol=1e-10)
+            and not np.isclose(abs(fit_trace - trace), 0, atol=1e-10)
+        ):
+            scale = trace / fit_trace
+            fit = fit * scale
+            eigvals = eigvals * scale
+        else:
+            trace = fit_trace
+
+        # Convert class of value
+        if qpt:
+            state = Choi(fit, input_dims=input_dims, output_dims=output_dims)
+        else:
+            state = DensityMatrix(fit, dims=output_dims)
+
+        metadata = {
+            "trace": trace,
+            "eigvals": eigvals,
+            "eigvecs": eigvecs,
+            "raw_eigvals": raw_eigvals,
+            "rescaled_psd": rescaled_psd,
+            "fitter_metadata": fitter_metadata or {},
+        }
+
+        states.append(state)
+        states_metadata.append(metadata)
+
+    # Compute the conditional probability of each component so that the
+    # total probability of all components is 1, and optional rescale trace
+    # of each component
+    total_trace = sum(fit_traces)
+    for i, (fit_trace, meta) in enumerate(zip(fit_traces, states_metadata)):
+        # Compute conditional component probability from the the component
+        # non-rescaled fit trace
+        meta["component_probability"] = fit_trace / total_trace
+        meta["component_index"] = i
+        if conditionals:
+            meta["component_conditional"] = conditionals[i]
+
+    return states, states_metadata
+
+
+def _state_eigensystem(state: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Compute the eigensystem of a fitted state.
+
+    The eigenvalues are returned as a real array ordered from
+    smallest to largest eigenvalues.
+
+    Args:
+        state: the fitted state matrix.
+
+    Returns:
+        A pair of (eigenvalues, eigenvectors).
+    """
+    evals, evecs = la.eigh(state)
+    # Truncate eigenvalues to real part
+    evals = np.real(evals)
+    # Sort eigensystem from largest to smallest eigenvalues
+    sort_inds = np.flip(np.argsort(evals))
+    return evals[sort_inds], evecs[:, sort_inds]
+
+
+def _make_positive(evals: np.ndarray, epsilon: float = 0) -> np.ndarray:
+    """Rescale a real vector to be non-negative.
+
+    This truncates any negative values to zero and rescales
```

Comment:
Strictly speaking this can truncate any value smaller than epsilon. You need to drop epsilon or change this description.